### DISCUSSION TITLE
Build fails in device when I have cache for simulator vice versa.

### DISCUSSION THREAD

**Discussion by muukii** [swiftyfinch/Rugby#41] [Q&A] [answered]
While Rugby creates a cache for a simulator or device, the caches would be locked by one of the archs.
If the caches are compatible simulator (x86_64), building for device will fail because the cache does not have artifacts for arm64.

<img width="650" alt="CleanShot 2021-08-04 at 13 55 48@2x" src="https://user-images.githubusercontent.com/1888355/128124279-cf040abc-595b-4485-92c2-b401007221ae.png">

I'm not sure yet Rugby how it work, but I guess it does rewriting xcconfig file that CocoaPods created.
So what if we write xcconfig with the following syntax that separates each sdk or archs, this issue might be fixed?

https://nshipster.com/xcconfig/#conditionalizing-build-settings

```
BUILD_SETTING_NAME[sdk=sdk] = value for specified sdk
BUILD_SETTING_NAME[arch=architecture] = value for specified architecture
BUILD_SETTING_NAME[config=configuration] = value for specified configuration
```

For example:

```
CONFIGURATION_BUILD_DIR[arch=x86_64] = ${PODS_ROOT}/../.rugby/build/Debug-iphoneos/CNIOAtomics
CONFIGURATION_BUILD_DIR[arch=arm64] = ${PODS_CONFIGURATION_BUILD_DIR}/CNIOAtomics
```

And Thank you for your super amazing tool!

**Comment by swiftyfinch** [comment, marked as answer]
Hello @muukii, thank you for the feature request and feedback about 🏈 Rugby!

There is a similar discussion #35, because your description has one drawback. I'll try to explain it.

If we build like `rugby -s ios`, we don't update other ARCHs caches.
So it can lead to misunderstanding situations and raise questions.
With your suggestion, Xcode would find caches, but the build can fail because the cache for some arch expired.
And unfortunately, we wouldn't get a clear message about this from Xcode.

Also, there already were few similar requests. You are not alone 👍🏻
I need time to dive into this problem and think about a clear API for all of us 💪🏻

**Reply by muukii** [reply]
Thanks.

I understood what happened in Xcode.
Then, Does Rugby suppose to commit build files into git in order to share with other team members?
I thought if the cache supports simulator and device and committing it into git, every team member in a project can build faster. but I'm not sure if LLDB works well with the cache that created another machine.

Previously, I've been working on fixing cocoapods-binary ([my fork](https://github.com/muukii/cocoapods-binary)).
Almost there, but LLDB did not work well with prebuilt binary.

**Reply by swiftyfinch** [reply]
I think you can commit the arch folder via **git-lfs** and **cache.yml** into git.
But yeah, I'm not sure about other build folders. You can try, but I think there can be issues.
It's just more complicated things related to **xcodebuild** process.

<img width="219" alt="Screenshot 2021-08-06 at 12 04 33" src="https://user-images.githubusercontent.com/64660122/128486100-67435aed-240c-4306-918c-246a68051d91.png">
